### PR TITLE
custom authority，schema can be assigned to proxy users

### PR DESCRIPTION
**Pull request by lwclover**
authority enhance，sample：
```
- !AUTHORITY
  users:
    - root@:root
    - user1@:user1
    - user1@127.0.0.1:user1
  provider:
    type: SCHEMA_PRIVILEGES_PERMITTED
    props:
      user-schema-mappings: root@=test, user1@127.0.0.1=db_dal_admin, user1@=test, user1@=db_dal_admin
```

when using root login from any host，you can use database test.
when using user1 login from 127.0.0.1，you can use database db_dal_admin.
when using user1 login from any other host，you can use database test and db_dal_admin.

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/schema/builder/SchemaPrivilegeBuilder.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                "user-schema-mappings configuration `%s` is invalid, the configuration format should be like `username@hostname=schema`", each));
+        Map<ShardingSphereUser, ShardingSpherePrivileges> result = new HashMap<>();
+        Map<ShardingSphereUser, Set<String>> userSchemaMappings = new HashMap<>();
+        Arrays.asList(mappings).stream().map(SchemaPrivilegeBuilder::convertSchemas).forEach(each -> merge(each, userSchemaMappings));
+        users.forEach(each -> result.put(each, new SchemaPrivilegesPermittedShardingSpherePrivileges(userSchemaMappings.getOrDefault(each, Collections.emptySet()))));
+        return result;
+    }
+    
+    /**
+     * convert schemas.
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/schema/builder/SchemaPrivilegeBuilder.java:63]: Please upper case the first letter of java doc

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/schema/builder/SchemaPrivilegeBuilder.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        String yamlUser = userSchemaPair[0];
+        String username = yamlUser.substring(0, yamlUser.indexOf("@"));
+        String hostname = yamlUser.substring(yamlUser.indexOf("@") + 1);
+
+        result.put(new ShardingSphereUser(username, "", hostname), userSchemaPair[1]);
+        return result;
+    }
+    
+    /**
+     * merge user schema to result.
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/schema/builder/SchemaPrivilegeBuilder.java:80]: Please upper case the first letter of java doc

File: shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/schema/model/privilege/SchemaPrivilegesPermittedShardingSpherePrivileges.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.shardingsphere.authority.model.PrivilegeType;
+import org.apache.shardingsphere.authority.model.ShardingSpherePrivileges;
+import org.apache.shardingsphere.authority.provider.natived.model.subject.SchemaAccessSubject;
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class SchemaPrivilegesPermittedShardingSpherePrivileges implements ShardingSpherePrivileges {
+
+    private Set<String> schemas;
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-authority/shardingsphere-infra-authority-common/src/main/java/org/apache/shardingsphere/authority/provider/schema/model/privilege/SchemaPrivilegesPermittedShardingSpherePrivileges.java:33]: Can the attribute `schemas` to be final?

**Review by tristaZero — APPROVED** [review]
@lwclover Hey, thanks for your first PR in this community, nice work. ;-)

**Comment by lwclover** [comment]
Give myself a thumbs up

**Comment by CrackerSuperior** [comment]
> authority enhance，sample：
> 
> ```
> - !AUTHORITY
>   users:
>     - root@:root
>     - user1@:user1
>     - user1@127.0.0.1:user1
>   provider:
>     type: SCHEMA_PRIVILEGES_PERMITTED
>     props:
>       user-schema-mappings: root@=test, user1@127.0.0.1=db_dal_admin, user1@=test, user1@=db_dal_admin
> ```
> 
> when using root login from any host，you can use database test. when using user1 login from 127.0.0.1，you can use database db_dal_admin. when using user1 login from any other host，you can use database test and db_dal_admin.

So, what if I want user1 to be able to use all the libraries?

**Pull request merged**